### PR TITLE
[#177094036] Added new API to support newsletters for the portals

### PR DESCRIPTION
**Pull request by infantesimone**
#### Description

This PR introduces an API to support newsletters for the portals.

#### List of Changes

1. Added new http Azure function that implements the following API:
```yaml
  "/newsletters/{id}/recipients":
    post:
      operationId: PostNewslettersEmails
      description: Add an email to newsletter
      parameters:
      - name: id
        in: path
        required: true
        description: newsletter id
        type: string
      - name: body
        in: body
        required: true
        schema:
          $ref: "#/definitions/RecipientRequest"
      responses:
        "200":
          description: Recipient added
        "400":
          description: Bad request
        "500":
          description: generic error           

    RecipientRequest:
      type: object
      required:
          - email
          - recaptchaToken
      properties:
        name:
          type: string
        email:
          type: string
          format: email
        recaptchaToken:
          type: string 
```  
2. Added the following env variables (to add in Azure environment):
    - `MAILUP_SECRET=f63XXXXXXXXX12480`
    - `MAILUP_USERNAME=mXXXXXX`
    - `MAILUP_PASSWORD=XXXXXXX`
    - `MAILUP_CLIENT_ID=26XXXXXXXXXXXXXXXXXXXXXXXe`
    - `RECAPTCHA_SECRET=6dddddd0P7N0dddddTdd`

#### Motivation and Context

The goal of the new API is to add a recipient to newsletters in Mailup if the Google Recaptcha Token and input email are valid.

#### Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

#### Checklist:

- [x] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

**Review by pasqualespica — APPROVED** [review]
Really great job 👍 🚀 
It could be useful to add some details , on how to test it manually or better add mocks to verify with local test.
But this could be done into new PR, this one already  contains a lot stuffs.

**Comment by infantesimone** [comment]
@pasqualespica Thanks for the point,  in a dedicated PR I have to create a mock to test easily 👍

File: io-functions-pay-portal/PostNewslettersRecipients/handler.ts [hunk 1]
```diff
@@ -0,0 +1,276 @@
...
+      fromPredicate(
+        ar => ar.success,
+        _ => new Error(`Error checking recaptcha`)
+      )
+    );
+
+const getMailupAuthTokenTask = (): TaskEither<Error, MailupAuthToken> =>
+  tryCatch(
+    () =>
+      fetchApi(`https://services.mailup.com/Authorization/OAuth/Token`, {
```
Comment (gunzip) [inline io-functions-pay-portal/PostNewslettersRecipients/handler.ts:153]: everytime you deal with hardcoded values I suggest to pass it as default values for method params, in this way you can easily override them in unit tests

Comment (infantesimone) [reply io-functions-pay-portal/PostNewslettersRecipients/handler.ts:153]: Thanks, i will refactor with the tests 👍

File: io-functions-pay-portal/PostNewslettersRecipients/handler.ts [hunk 2]
```diff
@@ -0,0 +1,276 @@
...
+        body: `grant_type=password&client_id=${config.MAILUP_CLIENT_ID}&client_secret=${config.MAILUP_SECRET}&username=${config.MAILUP_USERNAME}&password=${config.MAILUP_PASSWORD}`,
```
Comment (pp-ps) [inline io-functions-pay-portal/PostNewslettersRecipients/handler.ts:154]: I'm unconvinced, why a Resource Owner flow?
I would have probably picked a Client Credentials flow, I find it difficult to distinguish between a Client and a Resource Owner here.

Does a "grant_type=client_credentials" work, after you remove "username" and "password"?

Comment (infantesimone) [reply io-functions-pay-portal/PostNewslettersRecipients/handler.ts:154]: @pp-ps  Unfortunately, the _Authorization code grant_ described in the [documentation](https://help.mailup.com/display/mailupapi/Authenticating+with+OAuth+v2) cannot be used in this case (rest API), as it requires entering user and password on the _mailup_ page and then defining a callback url. In the _Resource Owner Password Grant_ described by the [documentation](https://help.mailup.com/display/mailupapi/Authenticating+with+OAuth+v2) , unfortunately, user and password is mandatory, however a client id is also passed that allows you to distinguish the  _Client_ and a _Resource Owner_.
This kind of authentication _grant_type=client_credentials_ it is not documented, but I can try to do an empirical test.

**Pull request merged**